src/main/java/core/basesyntax/serviceimpl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator<List<String>> {
+    @Override
+    public List<String> getReport() {
+        Map<String, Integer> fruitStorage = Storage.fruitStorage;
```

Comment:
Direct access to the Storage.fruitStorage map violates encapsulation. It would be better to provide a method in the Storage class to get the necessary data without exposing the internal map.